src/App.tsx

```diff
@@ -1,14 +1,58 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
+import { STATUSES, TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<string>(STATUSES.ALL);
+  const [query, setQuery] = useState<string>('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos().then(res => setTodoList(res));
+  }, []);
+
+  const filteredTodoListByStatus = (): Todo[] => {
+    switch (status) {
+      case STATUSES.COMPLETED:
+        return todoList.filter(todo => todo.completed);
+
+      case STATUSES.ACTIVE:
+        return todoList.filter(todo => !todo.completed);
+
+      case STATUSES.ALL:
+      default:
+        return todoList;
+    }
+  };
+
+  const getTodoList = () => {
+    const list = filteredTodoListByStatus();
+
+    if (query) {
+      return list.filter(item =>
+        item.title.toLocaleLowerCase().includes(query.toLocaleLowerCase()),
+      );
+    } else {
+      return list;
+    }
```

Comment:
```suggestion
      return list.filter(item =>
        item.title.toLocaleLowerCase().includes(query.toLocaleLowerCase()),
      );
    }
    
    return list;
```
- Move this function as well
- You don't need `else`